src/swtpm/utils.c

```diff
@@ -346,3 +348,21 @@ ssize_t read_eintr(int fd, void *buffer, size_t buflen)
         return n;
     }
 }
+
+/*
+ * get_socket_domain: Get the domain of a socket
+ *
+ * @fd: file descriptor of a socket
+ *
+ * Returns -1 in case an error occurred, the socket type otherwise
+ */
+int get_socket_domain(int fd)
+{
+    int optval;
+    socklen_t optlen = sizeof(optval);
+
+    if (getsockopt(fd, SOL_SOCKET, SO_DOMAIN, &optval, &optlen) < 0)
```

Comment:
SO_DOMAIN is not a standard option supported on all platforms where swtpm runs on.